ast/src/asm_analysis/mod.rs

```diff
@@ -64,10 +64,12 @@ pub struct InstructionDefinitionStatement {
 }
 
 #[derive(Clone, Debug)]
-pub struct LinkDefinitionStatement {
+pub struct LinkDefinition {
     pub source: SourceRef,
-    /// the flag which activates this link. Should be boolean.
-    pub flag: Expression,
+    /// the instruction flag, if this is an instruction link. Should be boolean.
```

Comment:
Can you explain what the difference between the two flags is?

Reply:
In the end, they are a single multiplied flag. I had to split them because I need to know this link comes from an instruction, and what that instruction is (the optional flag serves this purpose).

Reply:
Can you add the explanation to the code comment?